### PR TITLE
Qute: introduce RenderedResults assert API for tests

### PR DESCRIPTION
**Pull request by mkouba**
Related to https://github.com/quarkiverse/quarkus-renarde/pull/198.

### CODE REVIEW

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 1]
```diff
@@ -2654,6 +2654,12 @@ The configration value is a regular expression that matches the template path re
 For example, `quarkus.qute.dev-mode.no-restart-templates=templates/foo.html` matches the template `src/main/resources/templates/foo.html`.
 The matching templates are reloaded and only runtime validations are performed.
 
+=== Testing
+
+In the test mode, the rendering results of injected and type-safe templates are recorded in the managed `io.quarkus.qute.RenderedResults` which is registered as a CDI bean.
+You can inject `RenderedResults` in a test or any other CDI bean and assert the results.
+However, it's possible to set the `quarkus.qute.test-mode.record-rendered-results` configuration property to `false` to disable this feature.
```
[the remark this answers is not in the document]
Comment (FroMage) [reply docs/src/main/asciidoc/qute-reference.adoc:2661]: Oh, perhaps in continuous testing mode it grows infinitely?

**Review by FroMage — APPROVED** [review]
LGTM, great!

File: docs/src/main/asciidoc/qute-reference.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (FroMage) [reply docs/src/main/asciidoc/qute-reference.adoc:2661]: OK, fine

**Review by FroMage — APPROVED** [review]

**Comment by mkouba** [comment]
@gsmet Do you happen to know if `io.quarkus.it.jpa.postgresql.ImageMetricsITCase.verifyImageMetrics` is a flaky test?

**Comment by mkouba** [comment]
> @gsmet Do you happen to know if `io.quarkus.it.jpa.postgresql.ImageMetricsITCase.verifyImageMetrics` is a flaky test?

Hm, it seems that this test fails in other PRs as well.

I'm going to merge this one.

**Pull request merged**